packages/gatsby-plugin-graphql/src/webpack.ts

```diff
@@ -29,7 +29,9 @@ export const target = join(root, 'public', publicPath)
 
 const queryCode = ({ name, value, sha256Hash }: QueryNode) => `
 export const ${name} = {
-  query: ${isProduction ? undefined : JSON.stringify(value)},
+  query: process.env.NODE_ENV === 'production' ? undefined : ${JSON.stringify(
+    value
+  )},
```

Comment:
what's the difference?

Reply:
Before, I've run the conditional inside webpack's hook and write different versions of the file depending on the build environment. 

Now, I always write the file with the conditional. The trick is that this conditional is tree shaked by webpack and only one  statement is added to the final bundle